### PR TITLE
Dummy feature module

### PR DESCRIPTION
**Pull request by DamnClin**
Close https://github.com/jhipster/jhipster-lite/issues/2444
Close https://github.com/jhipster/jhipster-lite/issues/2541

### CODE REVIEW

**Comment by DamnClin** [comment]
In fact this also needs https://github.com/jhipster/jhipster-lite/issues/2351

So I'll integrate: 
* https://github.com/jhipster/jhipster-lite/issues/1800
* https://github.com/jhipster/jhipster-lite/issues/2001

But that means it is after https://github.com/jhipster/jhipster-lite/issues/2467 (so sad)

File: src/main/java/tech/jhipster/lite/generator/server/springboot/mvc/dummy/domain/DummyFeatureModuleFactory.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          .template("RestBeerToCreate.java")
+          .and()
+         .add(
+           MAIN_SOURCE.append(SECONDARY).template("InMemoryBeersRepository.java"),
+           mainDestination.append(SECONDARY).append("InMemoryBeersRepository.java")
+         )
+        .batch(TEST_SOURCE, toSrcTestJava().append(packagePath))
+          .template("BeanValidationAssertions.java")
+          .template("BeanValidationTest.java")
+          .template("JsonHelper.java")
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/mvc/dummy/domain/DummyFeatureModuleFactory.java:81]: Remove

File: tests-ci/generate.sh [hunk 2]
```diff
@@ -104,13 +104,13 @@ elif [[ $application == 'fullapp' ]]; then
...
   callApi "/api/servers/spring-boot/api-documentations/springdoc/init-with-security-jwt"
   callApi "/api/servers/spring-boot/logs-spy"
-  callApi "/api/servers/spring-boot/features/dummy"
   callApi "/api/servers/hexagonal-architecture-documentation"
   callApi "/api/servers/bean-validation-test"
 
   callApi "/api/servers/spring-boot/databases/postgresql"
   callApi "/api/servers/spring-boot/features/user/postgresql"
   callApi "/api/servers/spring-boot/component-tests/cucumber"
+  callApi "/api/servers/spring-boot/features/dummy"
```
Comment (DamnClin) [inline tests-ci/generate.sh:113]: Also add in OAuth2 app

**Comment by pascalgrimaud** [comment]
Moving the bounty to the ticket

**Comment by DamnClin** [comment]
cc @pascalgrimaud for me it's ok, status check is stuck because...reasons but I think this is all good

**Comment by pascalgrimaud** [comment]
just be patient :)

**Comment by pascalgrimaud** [comment]
I'll test this feature tomorrow

File: src/main/resources/generator/server/springboot/mvc/dummy/dummy.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Dummy feature
+
+Looks like you added a dummy feature to your project...
+
+This feature is a simple business example and exercise for this kind of projects. Here's the business context:
+
+A brave beer retailler is handling his 3 shops with papers and pencils and he is starting to have some trouble scaling up. He asked us to a build a simple system where he'll be able to:
```
Comment (pascalgrimaud) [inline src/main/resources/generator/server/springboot/mvc/dummy/dummy.md:7]: retailer instead?

File: src/main/resources/generator/server/springboot/mvc/dummy/dummy.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+- Add new beers (with name and price);
+- Remove beers from selling;
+- Get a catalog of currently sold beers to display in the shops.
+
+Our brave team of 1 dev didn't build that on his own, in fact he was with the retailler (tasting the product) while doing the first version. As the goal was to show what can be done easily the persistence is in memory for now, but hopefully we'll be able to change that later.
```
Comment (pascalgrimaud) [inline src/main/resources/generator/server/springboot/mvc/dummy/dummy.md:13]: retailer?

File: src/main/resources/generator/server/springboot/mvc/dummy/dummy.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+Even this very simple product already helped the retailler (a lot)! So he came back to our team (with some beers!) to add a feature to manage orders. The team started something to allow orders taking but this time the retailler had to leave and the team was stuck! What's an order??? Is it something going on with a customer in the shop or is it something "done" received from somewhere else?
```
Comment (pascalgrimaud) [inline src/main/resources/generator/server/springboot/mvc/dummy/dummy.md:15]: retailer?

File: src/main/resources/generator/server/springboot/mvc/dummy/dummy.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+We got answer from the retailler later on but our team is currently recovering from a very serious headache. Can you end what they started on order considering that:
```
Comment (pascalgrimaud) [inline src/main/resources/generator/server/springboot/mvc/dummy/dummy.md:17]: retailer?

File: src/main/resources/generator/server/springboot/mvc/dummy/test/HttpSteps.java.mustache [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+import io.cucumber.java.en.Then;
+
+public class HttpSteps {
+
+  @Then("I should be forbidden")
+  public void shouldBeForbidden() {
+    assertThatLastResponse().hasHttpStatus(HttpStatus.FORBIDDEN);
+  }
+
```
Comment (pascalgrimaud) [inline src/main/resources/generator/server/springboot/mvc/dummy/test/HttpSteps.java.mustache:15]: Remove this extra line

File: src/main/resources/generator/server/springboot/mvc/dummy/test/beers-catalog.feature [hunk 8]
```diff
@@ -0,0 +1,30 @@
+Feature: Beers catalog
+
+  Scenario: Should not add beer as user
+    Given I am logged in as "user"
+    When I add beer to catalog
+      | Name       | Cloak of feathers |
+      | Unit price | 8.53              |
+    Then I should be forbidden
+
+  Scenario: Add beer to catalog
```
Comment (pascalgrimaud) [inline src/main/resources/generator/server/springboot/mvc/dummy/test/beers-catalog.feature:10]: Should add beer...

File: src/main/resources/generator/server/springboot/mvc/dummy/test/dummy/infrastructure/primary/beer/RestBeerTest.java.mustache [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+class RestBeerTest {
+
+  @Test
+  void shouldSerializeToJson() {
+    assertThat(JsonHelper.writeAsString(RestBeer.from(beer()))).isEqualTo(json());
+  }
+
+  static String json() {
+    return """
+        {\
```
Comment (pascalgrimaud) [inline src/main/resources/generator/server/springboot/mvc/dummy/test/dummy/infrastructure/primary/beer/RestBeerTest.java.mustache:20]: Not sure about my comment: is `\` needed here?

File: src/main/resources/generator/server/springboot/mvc/dummy/test/dummy/infrastructure/primary/beer/RestBeerToCreateTest.java.mustache [hunk 10]
```diff
@@ -0,0 +1,32 @@
+package {{packageName}}.dummy.infrastructure.primary.beer;
+
+import static {{packageName}}.BeanValidationAssertions.*;
+import static org.assertj.core.api.Assertions.*;
+
+import {{packageName}}.JsonHelper;
+import {{packageName}}.dummy.domain.beer.BeersFixture;
+import org.junit.jupiter.api.Test;
+
+class RestBeerToCreateTest {
```
Comment (pascalgrimaud) [inline src/main/resources/generator/server/springboot/mvc/dummy/test/dummy/infrastructure/primary/beer/RestBeerToCreateTest.java.mustache:10]: Missing UnitTest

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

File: src/main/resources/generator/server/springboot/mvc/dummy/test/dummy/infrastructure/primary/beer/RestBeerTest.java.mustache (hunk 9 above)
Comment (DamnClin) [reply src/main/resources/generator/server/springboot/mvc/dummy/test/dummy/infrastructure/primary/beer/RestBeerTest.java.mustache:20]: yep, without them we have line break and test fails

**Review by pascalgrimaud — APPROVED** [review]

**Comment by DamnClin** [comment]
Claimed https://opencollective.com/generator-jhipster/expenses/87078

**Comment by pascalgrimaud** [comment]
@DamnClin : approved

**Pull request merged**